packages/block-editor/src/components/block-list/index.native.js

```diff
@@ -492,17 +398,36 @@ const EmptyListComponentCompose = compose( [
 		const blockClientIds = getBlockOrder( rootClientId );
 		const insertionPoint = getBlockInsertionPoint();
 		const blockInsertionPointIsVisible = isBlockInsertionPointVisible();
-		const shouldShowInsertionPoint =
-			! isStackedHorizontally &&
-			blockInsertionPointIsVisible &&
-			insertionPoint.rootClientId === rootClientId &&
-			// If list is empty, show the insertion point (via the default appender)
-			( blockClientIds.length === 0 ||
-				// Or if the insertion point is right before the denoted block.
-				! blockClientIds[ insertionPoint.index ] );
 
 		return {
-			shouldShowInsertionPoint,
+			shouldShowInsertionPoint:
+				! isStackedHorizontally &&
+				blockInsertionPointIsVisible &&
+				insertionPoint.rootClientId === rootClientId &&
+				// If list is empty, show the insertion point (via the default appender)
+				( blockClientIds.length === 0 ||
+					// Or if the insertion point is right before the denoted block.
+					! blockClientIds[ insertionPoint.index ] ),
 		};
-	} ),
-] )( EmptyListComponent );
+	} );
+
+	if ( renderFooterAppender || renderAppender === false ) {
+		return null;
+	}
+
+	return (
+		<View style={ styles.defaultAppender }>
+			<ReadableContentView
+				align={
+					renderAppender ? WIDE_ALIGNMENTS.alignments.full : undefined
+				}
+			>
+				<BlockListAppender
+					rootClientId={ rootClientId }
+					renderAppender={ renderAppender }
+					showSeparator={ shouldShowInsertionPoint }
+				/>
+			</ReadableContentView>
+		</View>
+	);
```

Comment:
Should we update the usage of `ReadableContentView` in favor of the `useEditorWrapperStyles` hook? Or do we plan to do it in a separate PR?

Something like:

```suggestion
	const align = renderAppender ? WIDE_ALIGNMENTS.alignments.full : undefined;
	const [ wrapperStyles ] = useEditorWrapperStyles( { align } );

	if ( renderFooterAppender || renderAppender === false ) {
		return null;
	}

	const containerStyles = [ styles.defaultAppender, wrapperStyles ];

	return (
		<View style={ containerStyles }>
			<BlockListAppender
				rootClientId={ rootClientId }
				renderAppender={ renderAppender }
				showSeparator={ shouldShowInsertionPoint }
			/>
		</View>
	);
```

Reply:
> Should we update the usage of `ReadableContentView` in favor of the `useEditorWrapperStyles` hook? Or do we plan to do it in a separate PR?
> 
> Something like:

My preference is to keep this PRs focused solely on converting this file from a class to a functional component. My hope is that will make code review easier and avoid complex merge conflicts if we split future work between different contributors. 

I planned for additional PRs for [subtasks](https://github.com/wordpress-mobile/gutenberg-mobile/issues/5617#issuecomment-1532380331) and https://github.com/wordpress-mobile/gutenberg-mobile/issues/5618.

Reply:
Awesome! That sounds good, since I didn't see anything in the list mentioning `ReadableContentView` I thought I'd mentioned it. 👍 Thanks!